### PR TITLE
Add -appVersion and -appBuild parameters to specify versioning

### PR DESCRIPTION
**Pull request by andydotxyz**
Andorid, darwin and iOS will use version for the display version and build for the internal version id.
Windows will use a combined version which is <version>.<build>, i.e. 1.0.0.1.
These fields are optional for packaging (1.0 and 1 will be used like before otherwise)
but required when building a release (obvs!)

(replaces previously incomplete PR #755)

### Checklist:

- [x] Tests included. <- the actual package output has not been decoupled from the compile procees, these were run and verified
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: cmd/fyne/internal/mobile/manifest.go [hunk 1]
```diff
@@ -53,14 +53,16 @@ type manifestTmplData struct {
 	Name        string
 	Debug       bool
 	LibName     string
+	Version     string
+	Build       int
 }
 
 var manifestTmpl = template.Must(template.New("manifest").Parse(`
```
Comment (Jacalz) [inline cmd/fyne/internal/mobile/manifest.go:60]: Should this perhaps be moved out to a separate file in the internal template package?

File: cmd/fyne/internal/mobile/build_iosapp.go [hunk 2]
```diff
@@ -229,6 +232,8 @@ func iosCopyAssets(pkg *packages.Package, xcodeProjDir string) error {
 type infoplistTmplData struct {
 	BundleID string
 	Name     string
+	Version  string
+	Build    int
 }
 
 var infoplistTmpl = template.Must(template.New("infoplist").Parse(`<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Jacalz) [inline cmd/fyne/internal/mobile/build_iosapp.go:239]: Should this perhaps be moved out to a separate file in the internal template package?

**Review by Jacalz — COMMENTED** [review]
Two comments on possible improvements to code structure, not exactly problems with the PR.

File: cmd/fyne/internal/mobile/manifest.go (hunk 1 above)
Comment (andydotxyz) [reply cmd/fyne/internal/mobile/manifest.go:60]: These files are still a fork of the gomobile build tools, howeover loosely. Moving the content around will make it very difficult to apply any security / feature patches from upstream.
At some point we may choose to do that, but I don't think just now is that time.

Comment (Jacalz) [reply cmd/fyne/internal/mobile/manifest.go:60]: Good point. Let's no do that now.

Comment (andydotxyz) [reply cmd/fyne/internal/mobile/manifest.go:60]: So is this approved as that change is not needed?

**Review by Jacalz — APPROVED** [review]
Looks good to me! Nice work 👍

File: cmd/fyne/commands/package-darwin.go [hunk 3]
```diff
@@ -25,7 +27,10 @@ func (p *packager) packageDarwin() error {
 	info := filepath.Join(contentsDir, "Info.plist")
 	infoFile, _ := os.Create(info)
 
-	tplData := darwinData{Name: p.name, ExeName: exeName, AppID: p.appID}
+	tplData := darwinData{
+		Name: p.name, ExeName: exeName, AppID: p.appID,
+		Version: p.appVersion, Build: p.appBuild,
```
Comment (stuartmscott) [inline cmd/fyne/commands/package-darwin.go:32]: Please use either single line, or multi line, instead of this hybrid.

**Review by stuartmscott — CHANGES_REQUESTED** [review]

File: cmd/fyne/commands/package-darwin.go (hunk 3 above)
Comment (andydotxyz) [reply cmd/fyne/commands/package-darwin.go:32]: When new things like this come up that are not failures in the lint command we should really capture the requirement for our contributor documentation.

**Comment by andydotxyz** [comment]
Sorry @Jacalz the whitespace change invalidated your review

File: cmd/fyne/commands/package-darwin.go (hunk 3 above)
Comment (stuartmscott) [reply cmd/fyne/commands/package-darwin.go]: Good idea. Though there is a limit to how many things should be explicit for this project, vs how many are common Go idioms - eg we don't explicitly say not to end lines with semi-colon, even tho it is required by the language's formal grammar - luckily Go is nice enough to add missing semi-colons for us (https://medium.com/golangspec/automatic-semicolon-insertion-in-go-1990338f2649)

**Review by stuartmscott — APPROVED** [review]

File: cmd/fyne/commands/package-darwin.go (hunk 3 above)
Comment (andydotxyz) [reply cmd/fyne/commands/package-darwin.go]: It's not really the same as goimports will remove the trailing semicolon automatically.

If it is important enough to block a PR then I think it should be captured through the automatic tools, through Go docs or in our specs

**Review by Jacalz — APPROVED** [review]

File: cmd/fyne/commands/package-darwin.go (hunk 3 above)
Comment (stuartmscott) [reply cmd/fyne/commands/package-darwin.go]: Added to https://github.com/fyne-io/fyne/wiki/Contributing, feel free to tweak

**Comment by Stoyvo** [comment]
What is the correct syntax for using this appVersion and appBuild?

The following fails:
```
fyne package -release -os darwin -executable mybinary -appID "ABCD1234.com.example.myapp" -appVersion "1.2.3" -appBuild 2020111401 -certificate "developerID_application.cer" -profile "MyProfile" -icon "assets/icon.png"
```

Response:
```
flag provided but not defined: -appVersion
```

**Comment by Stoyvo** [comment]
@andydotxyz What is the correct syntax for using this appVersion and appBuild? It's not listed as a parameter in `fyne help package`

The following fails:
```
fyne package -release -os darwin -executable mybinary -appID "ABCD1234.com.example.myapp" -appVersion "1.2.3" -appBuild 2020111401 -certificate "developerID_application.cer" -profile "MyProfile" -icon "assets/icon.png"
```

Response:
```
flag provided but not defined: -appVersion
```

**Comment by andydotxyz** [comment]
Don't use "fyne package -release" use "fyne release". That mode will list the parameters

**Comment by Stoyvo** [comment]
@andydotxyz ah, understood. I misunderstood documentation for packaging. Thank you

**Comment by andydotxyz** [comment]
Not a problem - the docs may be wrong as we moved from "package -release" to "release".
Do you remember which page was misleading?

**Comment by Stoyvo** [comment]
@andydotxyz there was no reference to "release" while in the packaging documentation. One would argue that you package your application for release, and at least a reference to release documentation would be ideal: https://developer.fyne.io/started/distribution

**Comment by andydotxyz** [comment]
Thanks that is a good point @Stoyvo.
Out of interest did you follow documentation to add the "-release" flag (which I would also need to update)
or was it a case of finding it in the command and using it? (Should probably be deprecated now)

**Pull request merged**